### PR TITLE
Louis/cc bump v0.0.17

### PR DESCRIPTION
**Pull request by ldelossa**
This PR bumps CC to v0.0.17.
This CC version introduced a change in libindex's State method. Commits patch these differences.

### CODE REVIEW

File: indexer/httptransport.go [hunk 1]
```diff
@@ -99,7 +99,15 @@ func (h *HTTP) IndexReportHandler(w http.ResponseWriter, r *http.Request) {
...
 	}
 
-	validator := fmt.Sprintf(`"%s|%s"`, h.serv.State(), manifest.String())
+	state, err := h.serv.State(ctx)
+	if err != nil {
+		resp := &je.Response{
+			Code:    "internal error",
+			Message: "could not retrieve indexer state " + err.Error(),
+		}
+		je.Error(w, resp, http.StatusInternalServerError)
```
Comment (hdonnay) [inline indexer/httptransport.go:109]: need to return after this

File: indexer/httptransport.go [hunk 2]
```diff
@@ -204,15 +212,22 @@ func (h *HTTP) StateHandler(w http.ResponseWriter, r *http.Request) {
...
 	}
 	w.Header().Set("content-type", "application/json")
-	s := h.serv.State()
+	s, err := h.serv.State(ctx)
+	if err != nil {
+		resp := &je.Response{
+			Code:    "internal error",
+			Message: "could not retrieve indexer state " + err.Error(),
+		}
+		je.Error(w, resp, http.StatusInternalServerError)
```
Comment (hdonnay) [inline indexer/httptransport.go:223]: need to return after this

File: go.sum [hunk 3]
```diff
@@ -621,6 +623,7 @@ golang.org/x/xerrors v0.0.0-20190410155217-1f06c39b4373/go.mod h1:I/5z698sn9Ka8T
 golang.org/x/xerrors v0.0.0-20190513163551-3ee3066db522/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
 golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7 h1:9zdDQZ7Thm29KFXgAX/+yaf3eVbP7djjWp/dXAppNCc=
 golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
+golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543 h1:E7g+9GITq07hpfrRu66IVDexMakfv52eLZ2CXBWiKr4=
```
Comment (hdonnay) [inline go.sum:621]: where does this come from?

**Review by hdonnay — CHANGES_REQUESTED** [review]

File: go.sum (hunk 3 above)
Comment (ldelossa) [reply go.sum:621]: I'm not sure. let me run tidy.

**Review by hdonnay — APPROVED** [review]

**Pull request merged**